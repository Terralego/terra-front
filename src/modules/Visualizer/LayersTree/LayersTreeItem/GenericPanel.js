import React from 'react';
import classNames from 'classnames';

const GenericPanel = ({ visible, left, width, top, title, children }) => (
  <div
    className={classNames('filters-panel', {
      'filters-panel--visible': visible,
      'filters-panel--animate': true,
      'bp3-dark': true,
    })}
    style={{
      left: typeof left === 'number' && typeof width === 'number' && left + width,
    }}
  >
    <div
      className="filters-panel__arrow"
      style={{
        top,
      }}
    />
    <div className="filters-panel__content">
      <div className="filters-panel__header">
        <h2 className="filters-panel__title">{title}</h2>
      </div>
      <div className="filters-panel__form">{children}</div>
    </div>
  </div>
);

export default GenericPanel;
