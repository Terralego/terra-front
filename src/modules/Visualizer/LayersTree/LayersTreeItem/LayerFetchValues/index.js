import LayerFetchValues from './LayerFetchValues';
import { connectLayersTree } from '../../LayersTreeProvider/context';

export default connectLayersTree('fetchPropertiesValues')(LayerFetchValues);
