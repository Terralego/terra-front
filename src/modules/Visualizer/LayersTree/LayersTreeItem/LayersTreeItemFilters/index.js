import LayersTreeItemFilters from './LayersTreeItemFilters';
import { connectLayersTree } from '../../LayersTreeProvider/context';

export default connectLayersTree(({
  getLayerState, setLayerState, translate,
}, {
  layer, layer: { exclusive, layers },
}) => {
  const activeLayer = exclusive
    ? layers.find(l => getLayerState({ layer: l }).active) || layer
    : layer;

  return {
    filtersValues: getLayerState({ layer: activeLayer }).filters || {},
    setLayerState,
    activeLayer,
    translate,
  };
})(LayersTreeItemFilters);
