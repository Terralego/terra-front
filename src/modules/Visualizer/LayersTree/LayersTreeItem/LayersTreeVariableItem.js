import { Collapse } from '@blueprintjs/core';
import React, { useContext, useState } from 'react';
import LayersTreeItem from '.';
import Select from '../../../Forms/Controls/Select';
import context from '../LayersTreeProvider/context';

const mock_variables = [
  {
    id: 'fe241425-3e33-426a-b068-7bf81052b6e6',
    label: 'Zone géo',
  },
  {
    id: 'dbeae359-261a-43f7-8757-da9637ad48db',
    label: 'Année',
  },
];

const mock_layers = [
  {
    initialState: {
      active: false,
      opacity: 1,
    },
    widgets: {},
    id: 78,
    label: 'Toponymes',
    content: '',
    source_filter: '',
    layers: ['d5537b7e5fd1f0288e9d7e207ede5475'],
    legends: [],
    mainField: 'nom',
    variables: {
      'dbeae359-261a-43f7-8757-da9637ad48db': '2023',
      'fe241425-3e33-426a-b068-7bf81052b6e6': 'Département',
    },
    filters: {
      layer: 'geo_toponymes',
      layerId: 78,
      mainField: 'nom',
      fields: null,
      form: null,
      exportable: false,
    },
    source_credit: '',
  },
  {
    initialState: {
      active: false,
      opacity: 1,
    },
    widgets: {},
    variables: {
      'dbeae359-261a-43f7-8757-da9637ad48db': '2020',
      'fe241425-3e33-426a-b068-7bf81052b6e6': 'Région',
    },
    id: 76,
    label: 'Toponymes',
    content: '',
    source_filter: '',
    layers: ['f0c4d3276aa82b7d3c08c1029ace3f13'],
    legends: [],
    mainField: 'nom',
    filters: {
      layer: 'geo_toponymes',
      layerId: 76,
      mainField: 'nom',
      fields: null,
      form: null,
      exportable: false,
    },
    source_credit: '',
  },
  {
    initialState: {
      active: false,
      opacity: 1,
    },
    embed: [
      {
        src: 'http://perdu.com',
        title: 'dshfas',
        fullScreen: true,
      },
    ],
    variables: {
      'dbeae359-261a-43f7-8757-da9637ad48db': '2021',
      'fe241425-3e33-426a-b068-7bf81052b6e6': 'Département',
    },
    widgets: {},
    embed_enable: true,
    id: 61,
    label: 'Toponymes',
    content: '',
    source_filter: '',
    layers: ['dca13ca34c9756f511ac2ac49a4dfe89'],
    legends: [],
    mainField: 'nom',
    filters: {
      layer: 'geo_toponymes',
      layerId: 61,
      mainField: 'nom',
      fields: null,
      form: null,
      exportable: false,
    },
    source_credit: '',
  },
];

const getLayerVariables = layer =>
  layer.variables || (mock_layers.find(l => l.id === layer.id)?.variables ?? {});

const getSelectedLayer = (selectedVariables, layers) =>
  layers.find(layer =>
    Object.entries(selectedVariables).every(
      ([variable, value]) => getLayerVariables(layer)?.[variable] === value,
    ),
  );

const getValuesByVariable = (variables, layers) => {
  const values = {};
  variables.forEach(variable => {
    values[variable.id] = Array.from(
      new Set(layers.map(layer => getLayerVariables(layer)?.[variable.id])),
    ).sort();
  });
  return values;
};

const getPossibleValues = (variables, layers, selectedVariables) => {
  const possibleLayers = layers.filter(layer =>
    Object.entries(selectedVariables).every(
      ([variable, value]) => getLayerVariables(layer)?.[variable] === value,
    ),
  );
  return getValuesByVariable(variables, possibleLayers);
};

const LayersTreeVariableItem = ({ group, layers }) => {
  const variables = group.variables || mock_variables;

  const [selectedVariables, setSelectedVariables] = React.useState(
    getLayerVariables(layers[0]) || {},
  );

  const [layer, setLayer] = useState(getSelectedLayer(selectedVariables, layers));

  const { setLayerState, getLayerState, layersExtent, isDetailsVisible } = useContext(context);

  const valuesByVariable = getValuesByVariable(variables, layers);

  const handleChange = (variable, value) => {
    const newSelectedVariables = { ...selectedVariables, [variable]: value };
    const newLayer = getSelectedLayer(newSelectedVariables, layers);
    if (!newLayer) return;
    setLayerState({ layer, state: { active: false } });
    setLayerState({ layer: newLayer, state: { active: true } });
    setSelectedVariables(newSelectedVariables);
    setLayer(newLayer);

    // possible values for subset of selectedVariables
    console.log('possibleValues', getPossibleValues(variables, layers, { [variable]: value }));
  };

  const layerActive = getLayerState({ layer }).active;

  return (
    <>
      <div className="layerstree-group layerstree-group--active">
        {layer && (
          <LayersTreeItem
            key={layer.label}
            layer={layer}
            customLabel={
              layerActive
                ? `${layer.label} (${Object.values(selectedVariables).join(' - ')})`
                : null
            }
            extent={layersExtent?.[layer.label]}
            isDetailsVisible={isDetailsVisible}
          />
        )}
        <Collapse isOpen={layerActive}>
          <div style={{ padding: 5, display: 'flex', flexDirection: 'column', gap: '0.8rem' }}>
            {variables.map(variable => (
              <Select
                fullWidth
                value={selectedVariables[variable.id]}
                onChange={value => handleChange(variable.id, value)}
                values={valuesByVariable[variable.id]}
              />
            ))}
          </div>
        </Collapse>
      </div>
    </>
  );
};

export default LayersTreeVariableItem;
