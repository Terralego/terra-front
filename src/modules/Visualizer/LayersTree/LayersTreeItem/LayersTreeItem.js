import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import {
  Card,
  Switch,
  Elevation,
  Tag,
} from '@blueprintjs/core';
import uuid from 'uuid/v4';

import OptionsLayer from './OptionsLayer';
import LayersTreeItemFilters from './LayersTreeItemFilters';
import LayerProps from '../../types/Layer';
import LayersTreeItemOptions from './LayersTreeItemOptions';
import withDeviceSize from '../../../../hoc/withDeviceSize';
import WarningZoom from './WarningZoom';
import LayersTreeExclusiveItemsList from './LayersTreeExclusiveItemsList';
import Tooltip from '../../../../components/Tooltip';

export class LayersTreeItem extends React.Component {
  static propTypes = {
    layer: LayerProps.isRequired,
    isActive: PropTypes.bool,
    opacity: PropTypes.number,
    isTableActive: PropTypes.bool,
    total: PropTypes.number,
    setLayerState: PropTypes.func,
    isMobileSized: PropTypes.bool,
    isPhoneSized: PropTypes.bool,
  };

  static defaultProps = {
    isActive: false,
    opacity: 1,
    isTableActive: false,
    total: null,
    setLayerState () {},
    isMobileSized: false,
    isPhoneSized: false,
  }

  state = {
    isOptionsOpen: false,
    isFilterVisible: false,
    hasWidgetActive: false,
  }

  uuid = `toggle-${uuid()}`;

  componentWillUnmount () {
    this.resetFilterPanelListener();
  }

  onActiveChange = ({ target: { checked: active } }) => {
    const { layer, setLayerState } = this.props;
    setLayerState({ layer, state: { active, table: false } });
  }

  onOpacityChange = opacity => {
    const { activeLayer: layer, setLayerState } = this.props;
    setLayerState({ layer, state: { opacity } });
  }

  getFilterPanelRef = refs => {
    this.resetFilterPanelListener();
    document.body.addEventListener('mousedown', this.clickListener = ({ target }) => {
      if (refs.reduce((contains, el) => contains || el.contains(target), false)) {
        return;
      }
      this.setState({ isFilterVisible: false });
    });
  }

  handleOptionPanel = () => {
    const { isOptionsOpen } = this.state;
    this.setState({ isOptionsOpen: !isOptionsOpen });
  }

  toggleFilters = () => this.setState(({ isFilterVisible }) =>
    ({ isFilterVisible: !isFilterVisible }));

  toggleTable = () => {
    const { activeLayer: layer, isTableActive, setLayerState } = this.props;
    setLayerState({ layer, state: { table: !isTableActive } });
  };

  toggleWidgets = widget => () => {
    const { activeLayer: layer, widgets: prevWidgets = [], setLayerState } = this.props;
    const contains = this.isWidgetActive(widget);
    const { hasWidgetActive } = this.state;
    const widgets = [
      ...(contains
        ? prevWidgets.filter(w => w !== widget)
        : [...prevWidgets, widget]),
    ];

    this.setState({ hasWidgetActive: !hasWidgetActive });

    setLayerState({ layer, state: { widgets } });
  }

  isWidgetActive = widget => {
    const { widgets = [] } = this.props;
    return widgets.includes(widget);
  }

  resetFilterPanelListener () {
    if (this.clickListener) {
      document.body.removeEventListener('mousedown', this.clickListener);
    }
  }

  render () {
    const {
      layer,
      layer: {
        group,
        label = group,
        exclusive,
        filters: {
          form: layerForm, fields: layerFields,
        } = {},
      },
      activeLayer,
      activeLayer: {
        filters: { form = layerForm, fields = layerFields } = {},
        widgets = layer.widgets,
      } = {},
      isActive,
      opacity,
      isTableActive,
      total,
      hidden,
      isMobileSized,
      isPhoneSized,
      map,
    } = this.props;

    if (hidden) return null;

    const {
      isOptionsOpen, isFilterVisible, hasWidgetActive,
    } = this.state;
    const {
      onActiveChange,
      onOpacityChange,
      toggleFilters,
      toggleTable,
      getFilterPanelRef,
      toggleWidgets,
      handleOptionPanel,
      isWidgetActive,
    } = this;

    const totalResult = typeof (total) === 'number';
    const hasSomeOptionActive =
    isTableActive || isFilterVisible || isOptionsOpen || hasWidgetActive;

    const displayTableButton = fields && !!fields.length;

    return (
      <Card
        className={classnames('layerstree-node', { 'options-hidden': isActive })}
        elevation={Elevation.ZERO}
        style={{ opacity: isActive ? 1 : 0.7 }}
      >
        <div className={
          classnames(
            { 'layerstree-node-content': !isMobileSized },
            { 'layerstree-node-content--desktop--active': !isMobileSized && hasSomeOptionActive },
            { 'layerstree-node-content--mobile': isMobileSized },
          )
        }
        >
          <div className={
            classnames(
              { 'layerstree-node-content__item': !isMobileSized },
              { 'layerstree-node-content__item--mobile': isMobileSized },
            )
          }
          >
            <div className={
            classnames(
              { 'layerstree-node-content__item-label': !isMobileSized },
              { 'layerstree-node-content__item-label--mobile': isMobileSized },
            )
          }
            >
              <WarningZoom
                isActive={isActive}
                map={map}
                layer={layer}
              >
                <Switch
                  checked={!!isActive}
                  onChange={onActiveChange}
                  id={this.uuid}
                />
              </WarningZoom>
              <Tooltip
                content={label}
                hoverOpenDelay={2000}
                className="layerstree-node-content__item-label__tooltip"
              >
                <label className="layerstree-node-content__item-label__label" htmlFor={this.uuid}>{label}</label>
              </Tooltip>
              <div className="layerstree-node-content__item-label__total">
                {isActive && totalResult && (
                <Tag
                  intent="primary"
                  round
                >
                  {total}
                </Tag>
                )}
              </div>
            </div>
            {isMobileSized && isActive && exclusive && (
              <LayersTreeExclusiveItemsList
                layer={layer}
              />
            )}
          </div>
          {isActive && !isPhoneSized && (
          <LayersTreeItemOptions
            hasSomeOptionActive={hasSomeOptionActive}
            isOptionsOpen={isOptionsOpen}
            handleOptionPanel={handleOptionPanel}
            layer={activeLayer}
            toggleFilters={toggleFilters}
            isFilterVisible={isFilterVisible}
            getFilterPanelRef={getFilterPanelRef}
            form={form}
            toggleTable={toggleTable}
            isTableActive={isTableActive}
            displayTableButton={displayTableButton}
            toggleWidgets={toggleWidgets}
            widgets={widgets}
            isWidgetActive={isWidgetActive}
            map={map}
          />
          )}
        </div>
        {isOptionsOpen && isActive && (
          <OptionsLayer
            onOpacityChange={onOpacityChange}
            opacity={opacity}
          />
        )}
        <LayersTreeItemFilters layer={activeLayer} />
        {!isMobileSized && isActive && exclusive && (
          <LayersTreeExclusiveItemsList
            layer={layer}
          />
        )}
      </Card>
    );
  }
}

export default withDeviceSize()(LayersTreeItem);
