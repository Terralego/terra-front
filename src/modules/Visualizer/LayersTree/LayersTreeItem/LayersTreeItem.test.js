import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import LayersTreeProvider from '../LayersTreeProvider';
import { LayersTreeItem } from './LayersTreeItem';

jest.mock('@blueprintjs/core', () => ({
  Button () {
    return <p>Button</p>;
  },
  Card ({ children }) {
    return children;
  },
  Switch ({ className }) {
    return <p className={className}>Switch</p>;
  },
  Elevation: {
    TWO: 'two',
  },
  Tag () {
    return <p>Tag</p>;
  },
  Tooltip () {
    return <p>Tooltip</p>;
  },
  Intent: {
    WARNING: 'warning',
  },
  Menu () {
    return <p>Menu</p>;
  },
  Popover () {
    return <p>Popover</p>;
  },
  Radio () {
    return <p>Radio</p>;
  },
  RadioGroup () {
    return <p>RadioGroup</p>;
  },
  PopoverPosition: {},
}));

jest.mock('@blueprintjs/select', () => ({
  Select () {
    return <p>BPSelect</p>;
  },
}));

jest.mock('@blueprintjs/datetime', () => ({
  DateRangeInput () {
    return <p>DateRangeInput</p>;
  },
}));

jest.mock('uuid/v4', () => () => 'uuid');

jest.mock('./OptionsLayer', () => function OptionsLayer () {
  return <p>OptionsLayer</p>;
});

jest.mock('./LayersTreeItemFilters', () => function LayersTreeItemFilters () {
  return <p>LayersTreeItemFilters</p>;
});

jest.mock('react-dom', () => ({
  createPortal: node => node,
}));

jest.mock('./FiltersPanel', () => function FiltersPanel () {
  return <p>FiltersPanel</p>;
});

jest.mock('./LocateButton', () => () => <button type="button">Extent this layer</button>);


jest.mock('../../services/warningZoom', () => ({
  processWarningAccordingToZoom (map) {
    return { minZoomLayer: 14, showWarning: !!map };
  },
}));

it('should render correctly', () => {
  const tree = renderer.create((
    <LayersTreeProvider>
      <LayersTreeItem
        layer={{
          label: 'foo',
        }}
        isActive
        opacity={1}
      />
      <LayersTreeItem
        layer={{
          label: 'bar',
        }}
        opacity={0.3}
      />
      <LayersTreeItem
        layer={{
          label: 'sublayers',
          sublayers: [{
            label: 'sublayer 1',
          }, {
            label: 'sublayer 2',
          }],
        }}
        isActive
        opacity={1}
      />
      <LayersTreeItem
        layer={{ label: 'sublayers', filters: { form: [], fields: [] }, widgets: [{ component: 'foo' }] }}
        isActive
        isTableActive={false}
      />
      <LayersTreeItem
        layer={{ label: 'sublayers', displayed: false }}
      />
      <LayersTreeItem
        layer={{ label: 'sublayers', filters: { form: [], fields: [{ label: 'foo' }] } }}
        isActive
        isTableActive={false}
      />
      <LayersTreeItem
        layer={{ label: 'sublayers', filters: { form: [], fields: [{ label: 'foo' }] } }}
        isActive
        isTableActive
        total={42}
      />
      <LayersTreeItem
        layer={{ label: 'hidden' }}
        hidden
      />
      <LayersTreeItem
        layer={{ label: 'warning' }}
        isActive
        map
      />
      <LayersTreeItem
        layer={{
          group: 'group 1',
          exclusive: true,
          layers: [{
            label: 'layer 1',
          }, {
            label: 'layer 2',
          }],
        }}
        isActive
      />
      <LayersTreeItem
        layer={{
          group: 'group 1',
          exclusive: true,
          filters: {
            layer: 'foo',
            form: [],
          },
          layers: [{
            label: 'layer 1',
          }, {
            label: 'layer 2',
          }],
        }}
        isActive
      />
      <LayersTreeItem
        layer={{
          group: 'group 1',
          exclusive: true,
          layers: [{
            label: 'layer 1',
          }, {
            label: 'layer 2',
            filters: {
              layer: 'foo',
              form: [],
            },
          }],
        }}
        isActive
      />
      <LayersTreeItem
        layer={{
          group: 'group 1',
          exclusive: true,
          filters: {
            layer: 'foo',
            form: [],
          },
          layers: [{
            label: 'layer 1',
          }, {
            label: 'layer 2',
          }],
        }}
        isActive
        isMobileSized
      />
    </LayersTreeProvider>
  ));
  const a = tree.root.findAll(({ type }) => type === LayersTreeItem);
  a[5].instance.setState({ isFilterVisible: true });
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should change active state', () => {
  const layer = {};
  const setLayerState = jest.fn();
  const instance = new LayersTreeItem({
    layer,
    setLayerState,
  });
  instance.onActiveChange({
    target: {
      checked: true,
    },
  });
  expect(setLayerState).toHaveBeenCalledWith({
    layer,
    state: { active: true, table: false },
  });
  setLayerState.mockClear();

  instance.onActiveChange({
    target: {
      checked: false,
    },
  });
  expect(setLayerState).toHaveBeenCalledWith({
    layer,
    state: { active: false, table: false },
  });
});

it('should change opacity state', () => {
  const layer = {};
  const setLayerState = jest.fn();
  const instance = new LayersTreeItem({
    layer,
    activeLayer: layer,
    setLayerState,
  });
  instance.onOpacityChange(42);
  expect(setLayerState).toHaveBeenCalledWith({
    layer,
    state: { opacity: 42 },
  });
});

it('should open options panel', () => {
  const instance = new LayersTreeItem({});
  instance.setState = jest.fn();

  instance.state = {
    isOptionsOpen: false,
  };
  instance.handleOptionPanel();
  expect(instance.setState).toHaveBeenCalledWith({
    isOptionsOpen: true,
  });
  instance.setState.mockClear();

  instance.state = {
    isOptionsOpen: true,
  };
  instance.handleOptionPanel();
  expect(instance.setState).toHaveBeenCalledWith({
    isOptionsOpen: false,
  });
});

it('should display options panel', () => {
  const wrapper = shallow((
    <LayersTreeItem
      layer={{
        label: 'foo',
      }}
      isActive
    />
  ));
  expect(wrapper.find('OptionsLayer').length).toBe(0);
  wrapper.setState({ isOptionsOpen: true });
  expect(wrapper.find('OptionsLayer').length).toBe(1);
});

it('should reset filter panel on unmount', () => {
  const instance = new LayersTreeItem();
  instance.resetFilterPanelListener = jest.fn();
  instance.componentWillUnmount();
  expect(instance.resetFilterPanelListener).toHaveBeenCalled();
});

it('should get filtered panel ref', () => {
  const instance = new LayersTreeItem();
  instance.resetFilterPanelListener = jest.fn();
  instance.setState = jest.fn();
  const refs = [{
    contains: () => true,
  }, {
    contains: () => false,
  }];
  jest.spyOn(document.body, 'addEventListener');
  instance.getFilterPanelRef(refs);
  expect(instance.resetFilterPanelListener).toHaveBeenCalled();
  expect(document.body.addEventListener).toHaveBeenCalledWith('mousedown', instance.clickListener);
  instance.clickListener({});
  expect(instance.setState).not.toHaveBeenCalled();

  instance.getFilterPanelRef([{ contains: () => false }]);
  instance.clickListener({ });
  expect(instance.setState).toHaveBeenCalledWith({ isFilterVisible: false });
});

it('should toggle filters', () => {
  const instance = new LayersTreeItem();
  instance.setState = jest.fn();
  instance.toggleFilters();
  expect(instance.setState).toHaveBeenCalledWith(expect.any(Function));
  const callback = instance.setState.mock.calls[0][0];
  expect(callback({ isFilterVisible: true })).toEqual({ isFilterVisible: false });
  expect(callback({ isFilterVisible: false })).toEqual({ isFilterVisible: true });
});

it('should toggle table', () => {
  const setLayerState = jest.fn();
  const layer = 'foo';
  const isTableActive = true;
  const instance = new LayersTreeItem({ setLayerState, layer, activeLayer: layer, isTableActive });
  instance.toggleTable();
  expect(setLayerState).toHaveBeenCalledWith({ layer, state: { table: false } });
});

it('should toggle widgets', () => {
  const layer = 'foo';
  const widget = {};
  const setLayerState = jest.fn();
  const instance = new LayersTreeItem({ layer, activeLayer: layer, setLayerState });

  instance.toggleWidgets(widget)();
  instance.setState = jest.fn();
  expect(setLayerState).toHaveBeenCalledWith({
    layer,
    state: {
      widgets: [widget],
    },
  });

  instance.props.widgets = [widget];
  instance.toggleWidgets(widget)();
  expect(setLayerState).toHaveBeenCalledWith({
    layer,
    state: {
      widgets: [],
    },
  });
  expect(instance.setState).toHaveBeenCalledWith({ hasWidgetActive: true });
});

it('should reset filters panel listener', () => {
  const instance = new LayersTreeItem();
  jest.spyOn(document.body, 'removeEventListener');
  instance.resetFilterPanelListener();
  expect(document.body.removeEventListener).not.toHaveBeenCalled();
  instance.clickListener = {};
  instance.resetFilterPanelListener();
  expect(document.body.removeEventListener).toHaveBeenCalledWith('mousedown', instance.clickListener);
});
