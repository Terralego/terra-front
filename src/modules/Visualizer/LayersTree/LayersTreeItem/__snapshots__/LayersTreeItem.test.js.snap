// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <div
    className="layerstree-node-content"
  >
    <div
      className="layerstree-node-content__item"
    >
      <div
        className="layerstree-node-content__item-label"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        />
      </div>
    </div>
    <div
      className="layerstree-node-content__options layerstree-node-content__options--desktop"
    >
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <button
          type="button"
        >
          Extent this layer
        </button>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          Tooltip
        </p>
      </span>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
  <div
    className="layerstree-node-content"
  >
    <div
      className="layerstree-node-content__item"
    >
      <div
        className="layerstree-node-content__item-label"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        />
      </div>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
  <div
    className="layerstree-node-content"
  >
    <div
      className="layerstree-node-content__item"
    >
      <div
        className="layerstree-node-content__item-label"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        />
      </div>
    </div>
    <div
      className="layerstree-node-content__options layerstree-node-content__options--desktop"
    >
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <button
          type="button"
        >
          Extent this layer
        </button>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          Tooltip
        </p>
      </span>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
  <div
    className="layerstree-node-content"
  >
    <div
      className="layerstree-node-content__item"
    >
      <div
        className="layerstree-node-content__item-label"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        />
      </div>
    </div>
    <div
      className="layerstree-node-content__options layerstree-node-content__options--desktop"
    >
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <button
          type="button"
        >
          Extent this layer
        </button>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          Tooltip
        </p>
      </span>
      <p>
        Popover
      </p>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
  <div
    className="layerstree-node-content"
  >
    <div
      className="layerstree-node-content__item"
    >
      <div
        className="layerstree-node-content__item-label"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        />
      </div>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
  <div
    className="layerstree-node-content layerstree-node-content--desktop--active"
  >
    <div
      className="layerstree-node-content__item"
    >
      <div
        className="layerstree-node-content__item-label"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        />
      </div>
    </div>
    <div
      className="layerstree-node-content__options layerstree-node-content__options--desktop layerstree-node-content__options--active"
    >
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <button
          type="button"
        >
          Extent this layer
        </button>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          Tooltip
        </p>
      </span>
      <p>
        Popover
      </p>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
  <div
    className="layerstree-node-content layerstree-node-content--desktop--active"
  >
    <div
      className="layerstree-node-content__item"
    >
      <div
        className="layerstree-node-content__item-label"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        >
          <p>
            Tag
          </p>
        </div>
      </div>
    </div>
    <div
      className="layerstree-node-content__options layerstree-node-content__options--desktop layerstree-node-content__options--active"
    >
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <button
          type="button"
        >
          Extent this layer
        </button>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          Tooltip
        </p>
      </span>
      <p>
        Popover
      </p>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
  <div
    className="layerstree-node-content"
  >
    <div
      className="layerstree-node-content__item"
    >
      <div
        className="layerstree-node-content__item-label"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        />
      </div>
    </div>
    <div
      className="layerstree-node-content__options layerstree-node-content__options--desktop"
    >
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <button
          type="button"
        >
          Extent this layer
        </button>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          Tooltip
        </p>
      </span>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
  <div
    className="layerstree-node-content"
  >
    <div
      className="layerstree-node-content__item"
    >
      <div
        className="layerstree-node-content__item-label"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        />
      </div>
    </div>
    <div
      className="layerstree-node-content__options layerstree-node-content__options--desktop"
    >
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <button
          type="button"
        >
          Extent this layer
        </button>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          Tooltip
        </p>
      </span>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
  <p>
    RadioGroup
  </p>,
  <div
    className="layerstree-node-content"
  >
    <div
      className="layerstree-node-content__item"
    >
      <div
        className="layerstree-node-content__item-label"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        />
      </div>
    </div>
    <div
      className="layerstree-node-content__options layerstree-node-content__options--desktop"
    >
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <button
          type="button"
        >
          Extent this layer
        </button>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          FiltersPanel
        </p>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          Tooltip
        </p>
      </span>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
  <p>
    RadioGroup
  </p>,
  <div
    className="layerstree-node-content"
  >
    <div
      className="layerstree-node-content__item"
    >
      <div
        className="layerstree-node-content__item-label"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        />
      </div>
    </div>
    <div
      className="layerstree-node-content__options layerstree-node-content__options--desktop"
    >
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <button
          type="button"
        >
          Extent this layer
        </button>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          Tooltip
        </p>
      </span>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
  <p>
    RadioGroup
  </p>,
  <div
    className="layerstree-node-content--mobile"
  >
    <div
      className="layerstree-node-content__item--mobile"
    >
      <div
        className="layerstree-node-content__item-label--mobile"
      >
        <p>
          Tooltip
        </p>
        <div
          className="layerstree-node-content__item-label__total"
        />
      </div>
      <p>
        RadioGroup
      </p>
    </div>
    <div
      className="layerstree-node-content__options layerstree-node-content__options--desktop"
    >
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <button
          type="button"
        >
          Extent this layer
        </button>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          FiltersPanel
        </p>
      </span>
      <span
        className="layerstree-node-content__options__overflow--shown"
      >
        <p>
          Tooltip
        </p>
      </span>
    </div>
  </div>,
  <p>
    LayersTreeItemFilters
  </p>,
]
`;
